### PR TITLE
Fix gunmod checks

### LINKED ISSUE #649: Incorrect warning message on installing gunmods to guns with internal magazines
# Describe the bug

For weapons with an internal magazine, i.e. loaded with ammo directly instead of a detachable mag, installing gunmods on them displays the "this will force you to need a magazine adapter mod" warning, even for gunmods that don't seem to affect loading.

# Steps To Reproduce

1. Debug spawn in a mosin nagant or other firearm that's loaded directly.
2. Spawn in a shoulder strap, suppressor, whatever else will fit on your gun of choice.
3. Install the gunmod, making note of the warning message shown.
4. Unload and reload the gun to confirm it still accepts cartridges just fine.

# Expected behavior

Message presumably is only intended to appear with specific gunmods, my rough guess is the belt feed adapter or something like that, hard to parse exactly what its intended use case is.

Wording also make me think of how scopes or other mods might hypothetically hinder the use speedloaders with bolt-action weapons, but it seems to be implying the exact opposite far as I can determine.

# Screenshots

Initial screenshot of when I first noticed this:
image

A test screenshot with an M4A1, confirming that it seems unaffected by this:
image

Then a moist nugget, based on an inkling I had after the M4 turned out fine:
image

Finally, testing a different accessory:
image

# Versions and configuration

```
- OS: Windows
    - OS Version: 10.0 2004
- Game Version: 528bb72 [64-bit]
- Graphics Version: Tiles
- Game Language:  []
- Mods loaded: [
    Dark Days Ahead [dda],
    Arcana and Magic Items [Arcana],
    Medieval Mod Reborn [Medieval_Mod_Reborn],
    MST Extra [MST_Extra],
    Cataclysm++ [Cata++],
    Mining Mod [Mining_Mod],
    Fewer Farms [Fewer_Farms],
    Tankmod: Revived [Tankmod_Revived],
    No Fungal Monsters [No_Fungi],
    Disable NPC Needs [no_npc_food],
    Faster Stamina Regen [stamina_regen_buff],
    Manual Bionic Installation [manualbionicinstall],
    Nonperishable Overhaul [Nonperishable_Overhaul],
    Simplified Nutrition [novitamins]
]
```

# Additional context

I'm still not sure what a magazine adapter mod is in this context, and why a separate gunmod would mandate the installation of a different one, aside from maybe some first stab at a multi-part caliber conversion that requires the player to change the receiver+barrel separate from the magazine well?

### PR DESCRIPTION
**Pull request by KheirFerrum**
#### Summary

SUMMARY: Bugfixes "Fix gunmod installation checks for integral magazines."

#### Purpose of change

- fixes #649

When gunmods are installed, warning for magazine adaptor appears in inappropriate situations. As the check does nothing within the same universe as what the warning is meant to indicate. It checks whether the gun's loaded magazine has a default ammo type. Meaning that if you could install a conversion kit while the gun was loaded (which is impossible), the check would pass.

Furthermore, it fails due to guns with integral magazines not returning a default ammo type, as they count as "not loaded" with a magazine. It also fails when unloaded guns are modified, even when the modification doesn't change the ammo type used by the gun.

#### Describe the solution

Rewrite the entire check so it provides appropriate warning.

#### Describe alternatives you've considered

- Make the code only check if ammo type is changed, stating that a magazine adaptor _may_ be required.
  - Would be a hell of a lot simpler, but I have standards.

#### Testing
- [x] Spawn M4A1, remove magazine, install any compatible gunmod that is not a conversion kit: waterproofing, shoulderstrap, suppressor, etc. Check that no warning fires.
  - [x] With magazine in, install any compatible gunmod, check that no warning fires.
- Via Json, add Bore mod location to Remington 1100.
  - [x] Spawn Remington 1100 shotgun, install any compatible gunmod that is not a conversion kit. Check that no warning fires.
- [x] Spawn 7.62x51mm caliber conversion kit, install into M4A1 and Remington, check that warning only fires for M4A1.
- [x] Spawn .300 AAC Blackout AR-15 conversion kit. Install into M4A1, check that no warning fires.

#### Additional context

### CODE REVIEW

**Comment by scarf005** [comment]
could you add instructions on how to test the PR?

**Comment by chaosvolt** [comment]
Related issue: https://github.com/cataclysmbnteam/Cataclysm-BN/issues/649

Testing the main issue this seeks to fix, as quoted from the bug report:
```
1. Debug spawn in a mosin nagant or other firearm that's loaded directly.
2. Spawn in a shoulder strap, suppressor, whatever else will fit on your gun of choice.
3. Install the gunmod, making note of the warning message shown.
4. Unload and reload the gun to confirm it still accepts cartridges just fine.
```
If fixed, step three will differ in there being no warning message displayed.

Testing whether conversion kits still work SHOULD still be doable by spawning them in debug, as I found them still buried in json/obsoletion/items.json so search the item spawn menu for `conversion kit`

**Comment by KheirFerrum** [comment]
> Related issue: #649
> 
> Testing the main issue this seeks to fix, as quoted from the bug report:
> 
> ```
> 1. Debug spawn in a mosin nagant or other firearm that's loaded directly.
> 2. Spawn in a shoulder strap, suppressor, whatever else will fit on your gun of choice.
> 3. Install the gunmod, making note of the warning message shown.
> 4. Unload and reload the gun to confirm it still accepts cartridges just fine.
> ```
> 
> If fixed, step three will differ in there being no warning message displayed.
> 
> Testing whether conversion kits still work SHOULD still be doable by spawning them in debug, as I found them still buried in json/obsoletion/items.json so search the item spawn menu for `conversion kit`

My concern there is mainly that since they've been obsoleted for a while they may no longer be compatible. Furthermore, I'm not even sure what magazine adaptors exist and where we would test that.

**Review by scarf005 — APPROVED** [review]
<details><summary>Confirmed that the tests work:</summary>

waterproofings were used for testing all non-conversion gunmods.

> Spawn M4A1, remove magazine, install any compatible gunmod that is not a conversion kit: waterproofing, shoulderstrap, suppressor, etc. Check that no warning fires. 

image

> With magazine in, install any compatible gunmod, check that no warning fires.

image

> Via Json, add Bore mod location to Remington 1100
> Spawn Remington 1100 shotgun, install any compatible gunmod that is not a conversion kit. Check that no warning fires

image

>  Spawn 7.62x51mm caliber conversion kit, install into M4A1 and Remington, check that warning only fires for M4A1.

image


image

> Spawn .300 AAC Blackout AR-15 conversion kit. Install into M4A1, check that no warning fires.

image

image

</details> 

<details><summary>.300 BLK M4A1 can still be loaded with 5.56x45mm rounds (and it won't fire)</summary>

image

image

image

</details> 

in the future we should prevent reloading magazines with incompatible ammos


<details><summary>Testing conversion kit with shotguns</summary>


image

image

image

while the gunmods installation works, there's still 00 shot left after installing conversion kit to remington 1100. but i guess it's okay as the game does not let you shoot while incompatible ammos are loaded

</details> 

<details><summary>#649</summary>

image

</details> 

LGTM
